### ISSUE TITLE
Intrinsic gas too low Error while deploying the contract

### ISSUE THREAD

**Issue by poprygun** [0xPolygon/polygon-edge#150]
### Description

Getting an error using hardhat contract deployment procedure.  I followed the steps in [Getting Started Guide](https://sdk-docs.polygon.technology/docs/how-tos/howto-setup-ibft/howto-set-ibft-locally), and was able to start all nodes without any errors, however, I am not able to deploy a simple contract.

Contract code:

```javascript
//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.0;

contract Greeter {
  string greeting;

  constructor(string memory _greeting) {
    greeting = _greeting;
  }

  function greet() public view returns (string memory) {
    return greeting;
  }

  function setGreeting(string memory _greeting) public {
    greeting = _greeting;
  }
}

```

hardhat.config.js section with network:

```javascript
module.exports = {
  solidity: "0.8.0",
  defaultNetwork: "polygon",
  networks: {
    polygon: {
      chainId: 100,
      url: 'http://localhost:30002',
      accounts: ['0x${private-key-hardcoded}'],
      gas: "auto",
      gasPrice: "auto",
      timeout: 20000
    }
  }
};
```

deploy.js script:

```javascript
async function main() {
  // We get the contract to deploy
  const [deployer] = await ethers.getSigners();

  console.log("Deploying contracts with the account:", deployer.address);

  console.log("Account balance:", (await deployer.getBalance()).toString());

  const Greeter = await ethers.getContractFactory("Greeter");
  const greeter = await Greeter.deploy("Hello, Hardhat!");

  console.log("Greeter deployed to:", greeter.address);
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```

### Your environment

* macOS Catalina
* version of the Polygon SDK
* current [develop branch](https://github.com/0xPolygon/polygon-sdk.git)

### Steps to reproduce

* Tell us how to reproduce this issue <br />

- Follow the steps  in [Getting Started Guide](https://sdk-docs.polygon.technology/docs/how-tos/howto-setup-ibft/howto-set-ibft-locally) to configure and run polygon-sdk locally
- Configure  sample hardhat project (see hardhat.config.js code above)
- Create sample contract (see provided source above)
- Run `npx hardhat run scripts/deploy.js --network polygon`

* Where the issue is, if you know <br />

* Which commands triggered the issue, if any

- attempt to deploy the contract

### Expected behaviour

Contract deployed and address returned

### Actual behaviour

"intrinsic gas too low"

### Logs

```
2021-09-09T16:21:30.670-0400 [ERROR] polygon.dispatcher: failed to dispatch: method=eth_sendRawTransaction err="intrinsic gas too low"
2021-09-09T16:25:22.933-0400 [ERROR] polygon.txpool: Discarding invalid transaction: hash=0x5e171beb08bfe8b6c5d645a5cec28c9f65bd19feda5212e12fd97f167ed04743 err="intrinsic gas too low"
2021-09-09T16:25:22.933-0400 [ERROR] polygon.dispatcher: failed to dispatch: method=eth_sendRawTransaction err="intrinsic gas too low"
```

### Proposed solution

If you have an idea of how to fix this issue, please write it down here, so we can begin discussing it

**Comment by zivkovicmilos** [comment]
Hey @poprygun ,

Thank you for reaching out and for bringing this to our attention. 
We've been able to reproduce the issue, and narrow it down to the root of it.

Currently, we're working on fixing it as soon as possible. I will update this issue when the fix PR is up so that you are in sync.

**Comment by zivkovicmilos** [comment]
One more thing, just so you can get up an running in the meantime while we sort this issue out:

You can bypass this issue and deploy your smart contract if you change your deploy params in the `deploy.js` file to be:
```js
const greeter = await Greeter.deploy("Hello, Hardhat!", {
    gasLimit: "0x100000",
  });
```

Notice that we specified the `gasLimit` here.

**Comment by zivkovicmilos** [comment]
Just to confirm here as well, this issue has been fixed now, and is awaiting [PR](https://github.com/0xPolygon/polygon-sdk/pull/154) approval 🚀

**Comment by poprygun** [comment]
Hi Milos,

I pulled latest develop and noticed few updates to server.go file, so I am assuming you merged the PR?

Adding below section causes script to hang indefinitely with no feedback from validator logs.

  const greeter = await Greeter.deploy("Hello, Hardhat!", {
    gasLimit: "0x100000",
  });

**Comment by poprygun** [comment]
Not sure if it is related, but creation of the contract using Remix/Metamask also hangs.

**Comment by zivkovicmilos** [comment]
Hey @poprygun ,

This fix still hasn't been merged into develop. Once the PR is merged (turns purple), you can pull the latest changes and the fix will be live.

Can you please share your entire hardhat project so I can see why it's hanging?
I have tested this out locally with both Hardhat and remix (with MM) and it works fine.

**Comment by poprygun** [comment]
Thanks, Milos,

Here is my hardhat project https://github.com/poprygun/oxy/tree/master/oxygenchain <https://github.com/poprygun/oxy/tree/master/oxygenchain>

Contract is hanging when I try to deploy from Remix IDE too...

**Comment by zivkovicmilos** [comment]
Thank you @poprygun for sharing the repo. I'll check to see if I can reproduce these issues you're facing and get back to you soon 🙂

**Comment by zivkovicmilos** [comment]
I've figured out what the issue is with your configuration.

You can safely omit the `gasPrice` and `gas` from the hardhat config, as they default to `auto`.

The polygon SDK (currently) has a max block limit of `0x500000` gas. You're setting the `gasLimit` value in your `deploy.js` to be `0x9000000`, which is way above what the actual block limit is. When you deploy the contract, the deployment transaction gets added to the transaction pool, but it doesn't get applied since you're hitting that gas limit. Because it was added to the txpool, hardhat got a returned txHash, which it uses to query the receipt with, causing the deploy script to hang until it eventually times out.

Simply setting this value to for example `0x100000` instead of `0x9000000` in `deploy.js`, and removing the `gasPrice` and `gas` params from the hardhat config should solve your issue and get you up and running.

**Comment by poprygun** [comment]
Thanks Milos,

I made suggested changes, however, still experiencing the same behavior.

Observation - And any value above ~ 0x13000 hangs the deploy process.

lowering gas limit in deploy.js to 0x13000 causes following log output on the node:

2021-09-16T08:57:12.734-0400 [ERROR] polygon.txpool: Discarding invalid transaction: hash=0x350bfa3f52fea52d8139a4662872b674161d31de06ab2cc9501d12db2895fed1 err="intrinsic gas too low"
2021-09-16T08:57:12.734-0400 [ERROR] polygon.dispatcher: failed to dispatch: method=eth_sendRawTransaction err="intrinsic gas too low”



I experience the somewhat similar behavior using remix/metamask with addition of this stack trace:

2021/09/16 09:13:21 http: panic serving 127.0.0.1:59138: runtime error: index out of range [-1]
goroutine 752 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1801 +0xb9
panic({0x1962920, 0xc00277dbd8})
	/usr/local/go/src/runtime/panic.go:1047 +0x266
github.com/0xPolygon/polygon-sdk/crypto.RecoverPubkey({0x0, 0x1, 0xc0000328f0}, {0xc000569ce0, 0x0, 0xc000569ca0})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/crypto/crypto.go:128 +0x135
github.com/0xPolygon/polygon-sdk/consensus/ibft.ecrecoverImpl({0x0, 0x0, 0x0}, {0xc000569ca0, 0x138b4fe, 0xc001722450})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/sign.go:22 +0xa5
github.com/0xPolygon/polygon-sdk/consensus/ibft.ecrecoverFromHeader(0x6d072f62c0cbfb31)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/sign.go:42 +0xc5
github.com/0xPolygon/polygon-sdk/consensus/ibft.(*Ibft).GetBlockCreator(0x7af5df61d56f1a33, 0xd92d2c3afaf35153)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/ibft.go:923 +0x2e
github.com/0xPolygon/polygon-sdk/server.(*jsonRPCHub).ApplyTxn(0xc001722c00, 0xc0021c8000, 0x4)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/server/server.go:302 +0x53
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Eth).Call(0xc00057e248, 0xc00146e2b0, 0xc00001f638)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/eth_endpoint.go:242 +0xa2
reflect.Value.call({0xc000246600, 0xc00057e2e8, 0x1976aa0}, {0x19c1503, 0x4}, {0xc002409f40, 0x3, 0x1842860})
	/usr/local/go/src/reflect/value.go:543 +0x814
reflect.Value.Call({0xc000246600, 0xc00057e2e8, 0xc000c0dce0}, {0xc002409f40, 0x3, 0x3})
	/usr/local/go/src/reflect/value.go:339 +0xc5
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Dispatcher).handleReq(0xc001720720, {{0x18710c0, 0xc000c0dce0}, {0xc00001f5f8, 0x8}, {0xc00256ec60, 0x8f, 0x90}})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/dispatcher.go:295 +0x309
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Dispatcher).Handle(0xc000c4c3c0, {0xc000c76000, 0xea, 0x200})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/dispatcher.go:242 +0x165
github.com/0xPolygon/polygon-sdk/jsonrpc.(*JSONRPC).handle(0xc001723140, {0x1cedf90, 0xc000b348c0}, 0xc00064b600)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/jsonrpc.go:218 +0x564
net/http.HandlerFunc.ServeHTTP(0x0, {0x1cedf90, 0xc000b348c0}, 0x10654a2)
	/usr/local/go/src/net/http/server.go:2046 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0, {0x1cedf90, 0xc000b348c0}, 0xc00064b600)
	/usr/local/go/src/net/http/server.go:2424 +0x149
net/http.serverHandler.ServeHTTP({0x1ceae08}, {0x1cedf90, 0xc000b348c0}, 0xc00064b600)
	/usr/local/go/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc00214c280, {0x1cf6000, 0xc000a340c0})
	/usr/local/go/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3033 +0x4e8


Also attached my genesis.json configuration

**Comment by poprygun** [comment]
Hi Milos,

I started the validator node in debug node, and captured this log output while attempting to deploy a contract - there are no errors, but node just cycles this output while hardhat is hanging deploying the contract.
Is there anything else we could try to troubleshot the deployment?


2021-09-18T10:42:07.994-0400 [DEBUG] polygon.dispatcher: request: method=eth_chainId id=513
2021-09-18T10:42:07.994-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":513,"result":"0x64"}
2021-09-18T10:42:07.996-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x73083785ffc1d90b41b3be9cf4ab090aeb54bb9744678cb7c87e95b305616e91"],"id":514}
2021-09-18T10:42:07.996-0400 [DEBUG] polygon.dispatcher: request: method=eth_getTransactionByHash id=514
2021-09-18T10:42:07.996-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":514,"result":null}
2021-09-18T10:42:08.000-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":515}
2021-09-18T10:42:08.000-0400 [DEBUG] polygon.dispatcher: request: method=eth_chainId id=515
2021-09-18T10:42:08.000-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":515,"result":"0x64"}
2021-09-18T10:42:08.002-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x73083785ffc1d90b41b3be9cf4ab090aeb54bb9744678cb7c87e95b305616e91"],"id":516}
2021-09-18T10:42:08.002-0400 [DEBUG] polygon.dispatcher: request: method=eth_getTransactionByHash id=516
2021-09-18T10:42:08.002-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":516,"result":null}
2021-09-18T10:42:12.006-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":517}
2021-09-18T10:42:12.006-0400 [DEBUG] polygon.dispatcher: request: method=eth_blockNumber id=517
2021-09-18T10:42:12.006-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":517,"result":"0x0"}
2021-09-18T10:42:12.007-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":518}
2021-09-18T10:42:12.007-0400 [DEBUG] polygon.dispatcher: request: method=eth_chainId id=518
2021-09-18T10:42:12.007-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":518,"result":"0x64"}
2021-09-18T10:42:12.010-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":519}
2021-09-18T10:42:12.010-0400 [DEBUG] polygon.dispatcher: request: method=eth_chainId id=519
2021-09-18T10:42:12.010-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":519,"result":"0x64"}
2021-09-18T10:42:12.012-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x73083785ffc1d90b41b3be9cf4ab090aeb54bb9744678cb7c87e95b305616e91"],"id":520}
2021-09-18T10:42:12.012-0400 [DEBUG] polygon.dispatcher: request: method=eth_getTransactionByHash id=520
2021-09-18T10:42:12.012-0400 [DEBUG] polygon.jsonrpc: handle: response={"jsonrpc":"2.0","id":520,"result":null}
2021-09-18T10:42:12.018-0400 [DEBUG] polygon.jsonrpc: handle: request={"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":521}

**Comment by tommyldev** [comment]
+1 also getting the same issue

**Comment by zivkovicmilos** [comment]
Hey @poprygun ,

I've merged out this fix into the develop branch.
Please pull the latest changes, and clean up any leftover files from the previous runs (any genesis.jsons, test-chain folders etc).

Let me know if the issue is still present after following these steps 🙂

**Comment by poprygun** [comment]
Hi Milos,

Unfortunately, latest change did not address my problem - contract deployment is still hanging from both - hardhat and remix.

However, deploying from Remix, I was able to capture this error on the node that is supposed to process transaction.


2021-09-20T10:20:26.350-0400 [DEBUG] polygon.jsonrpc: handle: request={"id":924458456,"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"from":"0x5e51889cac47fecc328bf2f32229035a037f04d3","data":"0x608060405234801561001057600080fd5b5060405161072c38038061072c8339818101604052810190610032919061015a565b806000908051906020019061004892919061004f565b50506102bf565b82805461005b9061022f565b90600052602060002090601f01602090048101928261007d57600085556100c4565b82601f1061009657805160ff19168380011785556100c4565b828001600101855582156100c4579182015b828111156100c35782518255916020019190600101906100a8565b5b5090506100d191906100d5565b5090565b5b808211156100ee5760008160009055506001016100d6565b5090565b6000610105610100846101cc565b61019b565b90508281526020810184848401111561011d57600080fd5b6101288482856101fc565b509392505050565b600082601f83011261014157600080fd5b81516101518482602086016100f2565b91505092915050565b60006020828403121561016c57600080fd5b600082015167ffffffffffffffff81111561018657600080fd5b61019284828501610130565b91505092915050565b6000604051905081810181811067ffffffffffffffff821117156101c2576101c1610290565b5b8060405250919050565b600067ffffffffffffffff8211156101e7576101e6610290565b5b601f19601f8301169050602081019050919050565b60005b8381101561021a5780820151818401526020810190506101ff565b83811115610229576000848401525b50505050565b6000600282049050600182168061024757607f821691505b6020821081141561025b5761025a610261565b5b50919050565b7f4e487b7100000000000000000000000000000000000000000000000000000000600052602260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052604160045260246000fd5b61045e806102ce6000396000f3fe608060405234801561001057600080fd5b50600436106100365760003560e01c8063a41368621461003b578063cfae321714610057575b600080fd5b6100556004803603810190610050919061022c565b610075565b005b61005f61008f565b60405161006c91906102a6565b60405180910390f35b806000908051906020019061008b929190610121565b5050565b60606000805461009e90610387565b80601f01602080910402602001604051908101604052809291908181526020018280546100ca90610387565b80156101175780601f106100ec57610100808354040283529160200191610117565b820191906000526020600020905b8154815290600101906020018083116100fa57829003601f168201915b5050505050905090565b82805461012d90610387565b90600052602060002090601f01602090048101928261014f5760008555610196565b82601f1061016857805160ff1916838001178555610196565b82800160010185558215610196579182015b8281111561019557825182559160200191906001019061017a565b5b5090506101a391906101a7565b5090565b5b808211156101c05760008160009055506001016101a8565b5090565b60006101d76101d2846102f9565b6102c8565b9050828152602081018484840111156101ef57600080fd5b6101fa848285610345565b509392505050565b600082601f83011261021357600080fd5b81356102238482602086016101c4565b91505092915050565b60006020828403121561023e57600080fd5b600082013567ffffffffffffffff81111561025857600080fd5b61026484828501610202565b91505092915050565b600061027882610329565b6102828185610334565b9350610292818560208601610354565b61029b81610417565b840191505092915050565b600060208201905081810360008301526102c0818461026d565b905092915050565b6000604051905081810181811067ffffffffffffffff821117156102ef576102ee6103e8565b5b8060405250919050565b600067ffffffffffffffff821115610314576103136103e8565b5b601f19601f8301169050602081019050919050565b600081519050919050565b600082825260208201905092915050565b82818337600083830152505050565b60005b83811015610372578082015181840152602081019050610357565b83811115610381576000848401525b50505050565b6000600282049050600182168061039f57607f821691505b602082108114156103b3576103b26103b9565b5b50919050565b7f4e487b7100000000000000000000000000000000000000000000000000000000600052602260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052604160045260246000fd5b6000601f19601f830116905091905056fea26469706673582212208c46b4a63bd9f426d83da8e4dcb69a8633659806f130d7a7ff99679dd5f5565b64736f6c6343000800003300000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000","value":"0x0"}]}
2021-09-20T10:20:26.350-0400 [DEBUG] polygon.dispatcher: request: method=eth_estimateGas id=9.24458456e+08
2021/09/20 10:20:26 http: panic serving 127.0.0.1:64966: runtime error: index out of range [-1]
goroutine 246 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1801 +0xb9
panic({0x1962d80, 0xc001c84480})
	/usr/local/go/src/runtime/panic.go:1047 +0x266
github.com/0xPolygon/polygon-sdk/crypto.RecoverPubkey({0x0, 0x1, 0x8}, {0xc001d93300, 0x0, 0xc001d932c0})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/crypto/crypto.go:128 +0x135
github.com/0xPolygon/polygon-sdk/consensus/ibft.ecrecoverImpl({0x0, 0x0, 0x0}, {0xc001d932c0, 0xc00052ed10, 0x100e454})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/sign.go:22 +0xa5
github.com/0xPolygon/polygon-sdk/consensus/ibft.ecrecoverFromHeader(0x0)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/sign.go:42 +0xc5
github.com/0xPolygon/polygon-sdk/consensus/ibft.(*Ibft).GetBlockCreator(0x0, 0x20)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/consensus/ibft/ibft.go:923 +0x2e
github.com/0xPolygon/polygon-sdk/server.(*jsonRPCHub).ApplyTxn(0xc000a41d40, 0xc000a706c0, 0x101010101010101)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/server/server.go:307 +0x53
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Eth).EstimateGas.func1(0x268d24)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/eth_endpoint.go:348 +0x68
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Eth).EstimateGas(0xc00027c2c0, 0xc001d980b0, 0x0)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/eth_endpoint.go:361 +0x9be
reflect.Value.call({0xc00043e980, 0xc00027c328, 0x1976f00}, {0x19c1963, 0x4}, {0xc000b0c5a0, 0x3, 0x1842cc0})
	/usr/local/go/src/reflect/value.go:543 +0x814
reflect.Value.Call({0xc00043e980, 0xc00027c328, 0xc001982748}, {0xc000b0c5a0, 0x3, 0x3})
	/usr/local/go/src/reflect/value.go:339 +0xc5
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Dispatcher).handleReq(0xc000a3d320, {{0x18658a0, 0xc001982748}, {0xc001982770, 0xf}, {0xc000d46000, 0xf29, 0x1000}})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/dispatcher.go:295 +0x485
github.com/0xPolygon/polygon-sdk/jsonrpc.(*Dispatcher).Handle(0xc000d3d000, {0xc000d3c000, 0xf6e, 0x1000})
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/dispatcher.go:242 +0x165
github.com/0xPolygon/polygon-sdk/jsonrpc.(*JSONRPC).handle(0xc000a722a0, {0x1cee4d0, 0xc0019782a0}, 0xc00050e800)
	/Users/ashumilov/projects/oxygenchain/polygon-sdk/jsonrpc/jsonrpc.go:218 +0x564
net/http.HandlerFunc.ServeHTTP(0x0, {0x1cee4d0, 0xc0019782a0}, 0x10654a2)
	/usr/local/go/src/net/http/server.go:2046 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0, {0x1cee4d0, 0xc0019782a0}, 0xc00050e800)
	/usr/local/go/src/net/http/server.go:2424 +0x149
net/http.serverHandler.ServeHTTP({0x1ceb348}, {0x1cee4d0, 0xc0019782a0}, 0xc00050e800)
	/usr/local/go/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc000ce4280, {0x1cf6540, 0xc0014ea0c0})
	/usr/local/go/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3033 +0x4e8
2021-09-20T10:20:26.546-0400 [DEBUG] polygon.jsonrpc: handle: request={"id":924458477,"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0x5b344c92f814a89c4a1c7aaeb025700c52de9eb62b26d449ae8d589402b28b22"]}

**Comment by poprygun** [comment]
Hi Milos, any progress on this?

Were you able to replicate the issue on your environment?

Thanks,

Alex

**Comment by poprygun** [comment]
One more note - I was able to deploy the contract when polygon was started in dev mode using `polygon-sdk dev` command, so something seems different in how cluster works vs “dev” mode.

**Comment by zivkovicmilos** [comment]
Hey @poprygun ,

I wasn't able to replicate this issue on my local setup, as well as other team members.

Could you please provide a detailed step by step guide as to how you're attempting this deployment?
Please note which configurations you use, what parameters you pass in, etc.

**Comment by poprygun** [comment]
I added the detailed description of the steps here https://github.com/poprygun/oxy/tree/master/oxygenchain <https://github.com/poprygun/oxy/tree/master/oxygenchain>

Note that I don’t have problem deploying when I start polygon with `polygon-sdk dev` command.

Thanks!

**Comment by Kourin1996** [comment]
Hello, @poprygun. Please note that you need to run 4 validator nodes at least in IBFT.

**Comment by poprygun** [comment]
That was it!  Needed to spin up a fourth node!

Thank you!
